packages/store/src/StoreCore.sol

```diff
@@ -259,6 +259,11 @@ library StoreCore {
       revert IStoreErrors.Store_InvalidResourceType(RESOURCE_TABLE, tableId, string(abi.encodePacked(tableId)));
     }
 
+    // Require the table to exist
+    if (!ResourceIds._getExists(tableId)) {
```

Comment:
are we sure we can use `_getExists` here over `getExists`?

if so, could the `StoreHooks.push` be `StoreHooks._push`?

Reply:
I think we can use the `_` because that calls `StoreCore`, which is already the contract we are in. [The other](https://github.com/latticexyz/mud/blob/main/packages/store/src/StoreCore.sol#L139) functions use it like this so I would follow this? It's not like `StoreCore` is delegatecalled.

Reply:
it feels out of scope of in this audit fix to change `StoreHooks._push` though

Reply:
let's make another PR for it?

Reply:
Here ya go: https://github.com/latticexyz/mud/pull/2201